src/pubsub/streaming_subscription.rs

```diff
@@ -442,25 +663,31 @@ where
     // the client id is used for stream reconnection on error
     let client_id = uuid::Uuid::new_v4().to_string();
 
-    // Channel to send/receive message ids to ack.
-    // This needs to be a multi-producer channel as each message will get a sender handle to send
-    // its ack. In practice the receiver will only ever be polled as single-consumer; however there
-    // is a period of time during reconnection when two receivers might exist, because the
-    // disconnected stream is dropped in a background task at an unknown time (unknown to our
-    // layer anyway), and the new receiver should continue to pull from the existing senders.
-    let (sender, receiver) = mpmc::bounded(
-        usize::try_from(config.max_outstanding_messages)
-            .expect("outstanding messages should fit in usize"),
-    );
+    let [mut client, ack_client, nack_client, modack_client] = clients;
 
     async_stream::stream! {
         let mut retry_op = None;
 
+        let (acks, acks_rx) = mpsc::unbounded_channel();
+        let (nacks, nacks_rx) = mpsc::unbounded_channel();
+        let (modacks, modacks_rx) = mpsc::unbounded_channel();
+        let ack_router = Arc::new(AckRouter { acks, nacks, modacks });
```

Comment:
Since `AckRouter` is just 3 senders, why not just clone it instead of putting it behind an `Arc`?